vuu-ui/packages/vuu-ui-controls/src/list/common-hooks/useKeyboardNavigation.ts

```diff
@@ -108,33 +111,29 @@ const pageDown = (
   }
 };
 
-const pageUp = async (
+const pageUp = (
   containerEl: HTMLElement,
   itemEl: HTMLElement,
   index: number
-): Promise<number | undefined> => {
+) => {
   const { top: itemTop } = itemEl.getBoundingClientRect();
   const { scrollTop, clientHeight } = containerEl;
   const newScrollTop = Math.max(scrollTop - clientHeight, 0);
-  if (newScrollTop !== scrollTop && index > 0) {
+  if (newScrollTop === scrollTop && index > 0) {
+    return 0;
+  } else if (newScrollTop !== scrollTop && index > 0) {
     containerEl.scrollTo(0, newScrollTop);
-    return new Promise((resolve) => {
-      // We must defer this operation until after render. If Items are virtualized.
-      // we need to allow them to be rendered.
-      requestAnimationFrame(() => {
-        let nextIdx = index;
-        let nextRect;
-        do {
-          nextIdx -= 1;
-          nextRect = getElementByDataIndex(
-            containerEl,
-            nextIdx,
-            true
-          ).getBoundingClientRect();
-        } while (nextRect.top > itemTop && nextIdx > 0);
-        resolve(nextIdx);
-      });
-    });
+    let nextIdx = index;
+    let nextRect;
+    do {
+      nextIdx -= 1;
+      nextRect = getElementByDataIndex(
+        containerEl,
+        nextIdx,
+        true
+      ).getBoundingClientRect();
+    } while (nextRect.top > itemTop && nextIdx > 0);
+    return nextIdx;
```

Comment:
Unrelated to this change, but (for the future) the flow would be easier to follow if we can have a separate function to calculate the next index. For example:
```
if (index > 0) {
  if (newScrollTop === scrollTop) {
    return 0;
  } else  {
    containerEl.scrollTo(0, newScrollTop);
    return next-idx-fn(); // should be a better name obv
}
```